lib/dialects/abstract/query-generator.js

```diff
@@ -1455,8 +1448,18 @@ var QueryGenerator = {
       if (options.where) {
         if (subQuery) {
           subQueryItems.push(' WHERE ' + options.where);
+
         } else {
           mainQueryItems.push(' WHERE ' + options.where);
+          // Walk the array backwards to find the last instance
+          for(var i = mainQueryItems.length - 1; i >= 0; --i) {
+            if(mainQueryItems[i].match(/^SELECT/)) {
+              var string = this.selectSqlFragmentBuilder(options, model, mainAttributes, table, mainTableAs, options.where);
+              //console.log(`Replacing '${mainQueryItems[0]}' with '${string}'`)
+              mainQueryItems[i] = string;
+              break;
+            }
```

Comment:
What happens here exactly? You attempt to find the first query item containg a select statement and then replace? Shouldn't there be an easier way of doing this.